### PR TITLE
[Py/Dims] error case to handle dimension @open sesame 09/01 11:29

### PR DESCRIPTION
**Pull request by jaeyun-jung**
Handle error case if rank from python script is larger than max rank value.
Update log message in python helper.

Signed-off-by: Jaeyun <jy1210.jung@samsung.com>

### CODE REVIEW

File: ext/nnstreamer/extra/nnstreamer_python3_helper.cc [hunk 1]
```diff
@@ -26,7 +26,8 @@
 /** @brief object structure for custom Python type: TensorShape */
 typedef struct
 {
-  PyObject_HEAD PyObject * dims;
+  PyObject_HEAD
```
Comment (myungjoo) [inline ext/nnstreamer/extra/nnstreamer_python3_helper.cc:29]: Actually it's...
```
#define PyObject_HEAD PyObject ob_base;
```

**Review by myungjoo — APPROVED** [review]

File: ext/nnstreamer/extra/nnstreamer_python3_helper.cc (hunk 1 above)
Comment (jaeyun-jung) [reply ext/nnstreamer/extra/nnstreamer_python3_helper.cc:29]: right, just indentation :)

File: ext/nnstreamer/extra/nnstreamer_python3_helper.cc [hunk 2]
```diff
@@ -397,72 +398,101 @@ int addToSysPath (const gchar *path)
...
     PyObject *tensor_shape = PyList_GetItem (result, (Py_ssize_t)i);
     if (nullptr == tensor_shape) {
-      Py_ERRMSG ("parseTensorsInfo() has failed (1).");
-      return -1;
-    }
-
-    PyObject *shape_dims = PyObject_CallMethod (tensor_shape, (char *)"getDims", NULL);
-    if (nullptr == shape_dims) {
-      Py_ERRMSG ("parseTensorsInfo() has failed (2).");
+      Py_ERRMSG ("parseTensorsInfo() has failed, cannot get TensorShape object.");
```
Comment (wooksong) [inline ext/nnstreamer/extra/nnstreamer_python3_helper.cc:411]: How about using ```__FUNCTION__``` or ```__PRETTY_FUNCTION__``` for better portability?

File: ext/nnstreamer/extra/nnstreamer_python3_helper.cc [hunk 3]
```diff
@@ -397,72 +398,101 @@ int addToSysPath (const gchar *path)
...
+        val = (int) PyFloat_AsDouble (item);
         ml_loge
-          ("Python nnstreamer plugin has returned the %d'th dimension value of the %u'th tensor in floating-point type (%f), which is casted as unsigned-int. Python code should return int-type for dimension values. Indexes are counted from 0.\n",
+          ("Python nnstreamer plugin has returned the %u'th dimension value of the %u'th tensor in floating-point type (%f), which is casted as unsigned-int. Python code should return int-type for dimension values. Indexes are counted from 0.\n",
           j, i, PyFloat_AsDouble (item));
       } else {
         info->info[i].dimension[j] = 0;
         ml_loge
-          ("Python nnstreamer plugin has returned the %d'th dimension value of the %u'th tensor neither in integer or floating-pointer. Python code should return int-type for dimension values. Indexes are counted from 0.\n",
+          ("Python nnstreamer plugin has returned the %u'th dimension value of the %u'th tensor neither in integer or floating-pointer. Python code should return int-type for dimension values. Indexes are counted from 0.\n",
```
Comment (wooksong) [inline ext/nnstreamer/extra/nnstreamer_python3_helper.cc:472]: Note: i and j start from 0. 
0'th, 1'th, 2'th, 3'th...

Comment (jaeyun-jung) [reply ext/nnstreamer/extra/nnstreamer_python3_helper.cc:472]: updated

File: ext/nnstreamer/extra/nnstreamer_python3_helper.cc (hunk 2 above)
Comment (jaeyun-jung) [reply ext/nnstreamer/extra/nnstreamer_python3_helper.cc]: updated

**Review by again4you — APPROVED** [review]
LGTM!

**Pull request merged**